filament/backend/src/opengl/OpenGLProgram.h

```diff
@@ -78,6 +96,13 @@ class OpenGLProgram : public HwProgram {
         GLuint program = 0;
     } gl;                                               // 4 bytes
 
+    PushConstantBundle getPushConstants() {
```

Comment:
what you could do instead, is have a `getPushConstants(ShaderStage)` and have it return a `utils::Slice< pair<GLint, ConstantType>>`, Slice is basically the same thing as `std::range` and it can be used like a vector, it's very convenient.  You would initialize it with `{ mPushConstants.data() + stageOffset, stageSize }` This essentially what you're doing already, except manually.
But the benefit would be that once you get the slice for a stage, there is no more overhead accessing a particular entry.

Reply:
The problem is I don't know which stage will be queried when the pipeline is bound (only when `setPushConstant` is called).  I could store both vertex and fragment slices in PushConstantBundle if you're ok with that.

Reply:
stored both slices in the bundle